clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -576,12 +576,8 @@ public void awaitFlushCompletion() throws InterruptedException {
         }
     }
 
-    public boolean hasUnflushedBatches() {
-        for (Map.Entry<TopicPartition, Deque<ProducerBatch>> entry : this.batches().entrySet()) {
-            if (!entry.getValue().isEmpty())
```

Comment:
Yes, I noticed this as well. If you look at the previous commits, I had one to fix it, but finally it seemed we didn't need it at all.